packages/cozy-authentication/src/utils/deeplink.js

```diff
@@ -12,8 +12,29 @@ const getDeeplink = () => {
   return window.deeplinkURL
 }
 
+const generateRoute = (url, appInfos) => {
+  const { protocol, universalLinkDomain, appSlug } = appInfos
+  const urlObj = new URL(url)
+
+  //remove fallback params not needed anymore at this stage
+  if (urlObj.searchParams.has('fallback')) {
+    urlObj.searchParams.delete('fallback')
+  }
+  let newUrl = urlObj.toString()
+  let path = ''
+  if (newUrl.startsWith('http')) {
+    path = newUrl.replace(universalLinkDomain + '/' + appSlug, '')
+  } else {
+    path = newUrl.replace(protocol, '')
+  }
+  if (path.startsWith('/')) path = path.substring(1)
```

Comment:
I think you can work with the urlObj a bit more without resorting to string replace that are a bit brittle. You can for example modify the protocol directly on the urlObj object `urlObj.protocol = ''` .